course/forms.py

```diff
@@ -93,3 +94,35 @@ def __init__(self, *args, **kwargs):
     class Meta:
         model = StudentGroup
         fields = ['members']
+
+
+class EnrollStudentsForm(forms.Form):
+
+    custom_errors = { 'invalid_choice': _('Selected users must be existing students who have not yet enrolled.')}
+    user_profiles = forms.ModelMultipleChoiceField(
+        queryset=UserProfile.objects.none(),
+        required=True,
+        error_messages=custom_errors
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.instance = kwargs.pop('course_instance')
+        super().__init__(*args, **kwargs)
+        self.fields['user_profiles'].widget.attrs['class'] = 'search-select'
+        self.fields['user_profiles'].queryset = UserProfile.objects.exclude(
+            id__in=[profile.id for profile in self.instance.get_student_profiles()]
+        )
+
+    def clean(self):
+        value = self.data.getlist('user_profiles')
+        existing_students = [i.id for i in self.instance.get_student_profiles()]
```

Comment:
Check existence with a database query. E.g. `Enrollment.objects.filter(course=, student=).exists()`

Reply:
Or you can retrieve all existing (`Enrollment.objects.filter(.., profile__in=[])`) and then return the error page.

Reply:
It would indeed be better that all the students didn't get rejected, if one was already in the course. The validation in the for loop seems sufficient, the students not in course could be added to the `cleaned_data` in some new value.